### ISSUE TITLE
Slow content resolution / fetching from LAN

### ISSUE THREAD

**Issue by swedneck** [ipfs/go-ipfs#7222]
#### Version information:
go-ipfs version: 0.5.0-rc4
Repo version: 9
System version: arm64/linux
Golang version: go1.13.10

#### Description:
Content resolution/fetching is slow, it takes upwards of 30 seconds to show an ipns page (created by a node on the LAN) on the http gateway.
This is on a rockpro64 running nextcloudpi, with apache reverse-proxying the http gateway port.

[ipfs-profile-nextcloudpi-2020-04-25T23_06_39+0200.tar.gz](https://github.com/ipfs/go-ipfs/files/4538326/ipfs-profile-nextcloudpi-2020-04-25T23_06_39%2B0200.tar.gz)

**Comment by willscott** [comment]
Are the LAN nodes behind the gateway able to connect out to the global IPFS network?

**Comment by swedneck** [comment]
yes, the node has plenty of connections and ipfs.io can fetch the content within a few seconds.

**Comment by Stebalien** [comment]
@dirkmc could you work with @swedneck to try to see where the bottleneck is?
